jsonnet/utils/generate-secret.libsonnet

```diff
@@ -21,6 +21,14 @@
               path: '/metrics',
               resourceRequest: false,
             },
+            {
```

Comment:
We only need prometheus to expose `/federate` endpoint to telemeter client. 
I think it better to limit the access to `/federate` endpoint to the `kube-rbac-proxy` before `prometheus-k8s` .
We can make a dedicated function to `prometheus-k8s` deployment or make a wrap function to add this config about `/federate` to the result of `staticAuthSecret`